packages/orchestration/src/util/tool-calls.ts

```diff
@@ -0,0 +1,72 @@
+import type {
+  MessageToolCall,
+  ToolCallChunk
+} from '../client/api/schema/index.js';
+
+type ToolCallAccumulator = {
+  id?: string;
+  type: 'function';
+  function: {
+    name?: string;
+    arguments?: string;
+  } & Record<string, any>;
+} & Record<string, any>;
+
+function validateToolCallAccumulator(acc: ToolCallAccumulator): boolean {
```

Comment:
[pp] Maybe write it as a type guard? Since later we write `return acc as MessageToolCall`, why not just return here `is MessageToolCall`?